interpret.py

```diff
@@ -49,48 +49,92 @@ class DebugFlow:
 	def __init__(self, interp = None):
 		self.interpreter = interp if interp else Interpreter()
 		self.interpreter.getoutput() # get out the python notice
-
-	def flow(self, code):
-		# For now we take the whole code since we might get invalid code so we
-		# could try to discard lines to have correct code.
-		print(code)
-		tree = ast.parse(code)
-		for node in ast.walk(tree):
-			if not isinstance(node, ast.Module) and \
-				not isinstance(node, ast.FunctionDef):
-				continue
-
+		self.seen = set()
+
+	def traverse(self, node, insideCall=False):
+		#TODO(rotaru): fix repetitive evaluation of the same name/expression
+		if node in self.seen:
+			return
+		else:
+			self.seen.add(node)
+
+		if isinstance(node, ast.Store) or \
```

Comment:
MAYBE: This is pretty ugly to check the instance for each ```ast``` type.
Can't we have an abstract class and inherit from it and overwrite a specific method? After that we could call that method, regardless of what that object is (we would move this ```if...else``` logic inside separate classes and I think is much cleaner.

I do not have a full picture about all of this - it might not be the correct approach.

Reply:
done